### ISSUE TITLE
[Accessibility]: Placeholder text in text input fields need to be more clarifying

### ISSUE THREAD

**Issue by Strum355** [sourcegraph/sourcegraph#35322]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/33514

### Problem description

It is unclear what e.g. "search matches" specifically means in certain text input fields. 

### Expected behavior

These should be clarified to reduce the expectation of visual association between the text and the remaining content.

### Additional details

![image](https://user-images.githubusercontent.com/18282288/167930773-1a98810d-7718-49cb-a9e1-4fef5e444ca5.png)

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by umpox** [comment]
@Strum355 Have assigned to GitStart, LMK if you disagree!

**Comment by gitstart-sourcegraph** [comment]
@umpox From ticket description, we need to update input placeholder of that search input on Code Intel Upload page. 
WDYT if we update `ConnectionFormProps` for enforcing `hideSearch=false` or `hideSearch=true` + required (inputAriaLabel or inputPlaceholder)?

**Comment by umpox** [comment]
@gitstart-sourcegraph Thanks for raising this, it actually seems that the particular issue has already been fixed. We have some logic now that sets the `ariaLabel` as `placeholder` if not specified otherwise, let's keep this logic and close this issue for now - will readdress if we have any more problems with it!

**Comment by gitstart-sourcegraph** [comment]
@umpox I think this issue still need changes to be closed. 

https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/web/src/enterprise/codeintel/uploads/pages/CodeIntelUploadPage.tsx?L348 
IMO this should be other words not `match` and `matches` here (This is reason we see `Search matches` on UI)
